src/async-wrap.cc

```diff
@@ -164,8 +168,6 @@ static void DestroyIdsCb(uv_idle_t* handle) {
       FatalException(env->isolate(), try_catch);
     }
   }
-
-  env->destroy_ids_list()->clear();
```

Comment:
Removing this fixes the issue tested by `parallel/test-async-hooks-close-during-destroy`. If somebody feels like I should, I can split it into a second commit.